pkg/controller/controller.go

```diff
@@ -692,6 +707,13 @@ func (s *Service) ControllerUnpublishVolume(ctx context.Context, req *csi.Contro
 	if !ok {
 		return nil, status.Errorf(codes.InvalidArgument, "cannot find array %s", arrayID)
 	}
+	var remoteArray *array.PowerStoreArray
+	if remoteArrayID != "" {
+		remoteArray, ok = s.Arrays()[remoteArrayID]
+		if !ok {
+			return nil, status.Errorf(codes.InvalidArgument, "cannot find remote array %s", remoteArrayID)
```

Comment:
Maybe we can change this to be a bit more descriptive. Something that indicates the secret wasn't configured.

e.g. `missing configuration for remote array %s`

Reply:
If we need to update the 'array' error messages, we will need to do it across the repository in a separate PR.